api/v1alpha1/hostedcluster_types.go

```diff
@@ -47,6 +48,22 @@ type ClusterNetworking struct {
 	MachineCIDR string `json:"machineCIDR"`
 }
 
+// ProviderCredentialsSpec describes cloud provider credentials used in various
+// contexts by HyperShift.
+type ProviderCredentialsSpec struct {
+	// KubeCloudControllerCreds is a reference to a secret containing cloud
+	// credentials with permissions matching the Kube cloud controller policy.
+	// The secret should have exactly one key, `credentials`, whose value is
+	// the platform specific credentials content.
+	KubeCloudControllerCreds corev1.LocalObjectReference `json:"kubeCloudControllerCreds"`
```

Comment:
I think it can be required as long as ProviderCreds inside HostedClusterSpec is not required. For platforms like baremetal these creds don't make sense.

Reply:
Are these really fields of AWSPlatformSpec?

Reply:
@ironcladlou they should be fields of AWSPlatformSpec if we think creds will be handled differently for other platforms. We should look at gcp and azure at least to understand if they require similar.

Reply:
Bare metal is another platform on which these fields are wholly inapplicable, isn't that enough?

Reply:
If ProviderCreds was optional, then it simply wouldn't be used in baremetal (or any other platform that doesn't require creds)

Reply:
Extending that argument, why wouldn't all fields from `PlatformSpec.AWS` be lifted up to HostedClusterSpec and made optional? If the credentials are coupled to the platform, wouldn't they belong on the field we already have for platform specific options (which are already optional via the `PlatformSpec.AWS` field)

Reply:
Because if we need the exact same fields for other clouds, then it's not the same as all the fields in PlatformSpec.AWS

Reply:
And, if now (need to do more R&D to determine) or in the future there is any difference between platforms for just the credentials spec, optionality won't be enough we'd end up having two parallel platform specific fields (Platform and ProviderCreds with its own discriminating union type for each provider's credentials struct).

Reply:
Ack, I'm fine with putting them in PlatformSpec.AWS